### PR TITLE
[FIX] Remove voip extension when agent is removed or deactivated

### PR DESCRIPTION
**Pull request by KevLehman**
## Further comments

Jira: OC-179

### CODE REVIEW

File: apps/meteor/app/livechat/server/hooks/afterUserActions.ts [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+};
+
+const handleDeactivateUser = async (userData: IUser) => {
+	if (userData?.roles?.includes('livechat-agent')) {
+		await Users.unsetExtension(userData._id);
+	}
+};
+
+callbacks.add(
+	'afterSaveUser',
```
[the remark this answers is not in the document]
Comment (KevLehman) [reply apps/meteor/app/livechat/server/hooks/afterUserActions.ts:22]: The thing is that, when an agent is removed (the actual user is removed) all seems to work 👀 . The problem is when you just remove _the role_ of the user, or when user is deactivated. 

But for my mental sanity, will test what happens when a user is removed. I remember we fixed that case some time ago.

[the remark this answers is not in the document]
Comment (murtaza98) [reply apps/meteor/app/livechat/server/hooks/afterUserActions.ts:22]: Oh... that makes sense :+1: Thanks for the prompt reply. This looks good to me then :)

File: apps/meteor/app/livechat/server/lib/Livechat.js [hunk 2]
```diff
@@ -946,6 +946,7 @@ export const Livechat = {
 			this.setUserStatusLivechat(_id, 'not-available');
 			LivechatDepartmentAgents.removeByAgentId(_id);
 			Promise.await(LivechatVisitors.removeContactManagerByUsername(username));
+			Promise.await(UsersRaw.unsetExtension(_id));
```
Comment (murtaza98) [inline apps/meteor/app/livechat/server/lib/Livechat.js:949]: Maybe we can combine this into one Promise all :arrow_up:

**Review by murtaza98 — APPROVED** [review]

**Review by murtaza98 — APPROVED** [review]

**Review by cauefcr — APPROVED** [review]
re-approving after update

**Pull request merged**